Sources/APNS/APNS.swift

```diff
@@ -0,0 +1,151 @@
+import Vapor
+
+extension Application {
+    public var apns: APNS {
+        .init(application: self)
+    }
+
+    public struct APNS {
+        final class Storage {
+            var pool: EventLoopGroupConnectionPool<APNSConnectionSource>?
+            init() { }
+        }
+
+        struct Key: StorageKey {
+            typealias Value = Storage
+        }
+
+        struct Lifecycle: LifecycleHandler {
+            func shutdown(_ application: Application) {
+                if let pool = application.apns.storage.pool {
+                    pool.shutdown()
+                }
+            }
+        }
+
+        var storage: Storage {
+            if self.application.storage[Key.self] == nil {
+                self.initialize()
+            }
+            return self.application.storage[Key.self]!
```

Comment:
Nit, but wouldn't this be cleaner?

```swift
        var storage: Storage {
            guard let existingStorage = self.application.storage[Key.self] else {
                self.initialize()
            }
            return existingStorage
        }
```

Reply:
`guard` must return before it exits scope so that wouldn't work

Reply:
sorry meant, 
```swift
        var storage: Storage {
            guard let existingStorage = self.application.storage[Key.self] else {
                return self.initialize()
            }
            return existingStorage
        }
```

but yeah, then initialize needs to return the key. So its just semantic.